### PR TITLE
fix: apply needed changes to deps and structure

### PR DESCRIPTION
**Pull request by peternhale**
@W-9368257@ Updates to plugin to bring it more in line with what tools are being used for other production plugins.

### CODE REVIEW

File: src/generators/app.ts [hunk 1]
```diff
@@ -241,11 +241,11 @@ class App extends Generator {
     const sfdxPluginOptions = {
       typescript: true,
       mocha: true,
-      tslint: true,
+      tslint: false,
```
Comment (mshanemc) [inline src/generators/app.ts:244]: yay!

File: src/generators/app.ts [hunk 2]
```diff
@@ -455,69 +456,88 @@ class App extends Generator {
...
+        '@salesforce/dev-config@^2',
+        '@salesforce/dev-scripts@^0',
+        '@salesforce/prettier-config@^0',
+        '@salesforce/ts-sinon@^1',
+        '@types/jsforce@^1.9.29',
+        '@typescript-eslint/eslint-plugin@^4',
+        '@typescript-eslint/parser@^4',
+        'eslint-config-prettier@^8',
+        'eslint-config-salesforce@^0',
+        'eslint-config-salesforce-license@^0',
```
Comment (mshanemc) [inline src/generators/app.ts]: plugin-generator is intended for customer use...let's not put our copyright on it by default

**Review by mshanemc — APPROVED** [review]

File: src/generators/app.ts (hunk 2 above)
Comment (peternhale) [reply src/generators/app.ts]: @mshanemc I'll remove salesforce license, what about our config?

File: src/generators/app.ts [hunk 3]
```diff
@@ -380,13 +381,13 @@ class App extends Generator {
         }
       }
     }
-    if (this.eslint) {
+    if (this.eslint && this.type !== 'sfdx-plugin') {
```
Comment (RodEsp) [inline src/generators/app.ts:384]: Please excuse my ignorance but what does this do?
Does it make it so our standard `eslintrc` isn't added to our own plugins?

Comment (peternhale) [reply src/generators/app.ts:384]: @RodEsp This repo was a direct clone of the oclif version of plugin generate. The oclif version asked a bunch of question, such as JS or TS, yarn or npm, ... The interactive piece of this plugin was stripped down to be sfdx specific, including language === 'ts', packager mgr === 'yarn', etc. However, all of the logic that consumes the answers were not stripped down, but left in place. Logic like `this.type !== 'sfdx-plugin'` is looking for things to for a sfdx plugin.

File: src/generators/app.ts [hunk 4]
```diff
@@ -274,15 +274,16 @@ class App extends Generator {
       this.pjson.scripts.posttest = 'eslint .'
     }
     if (this.mocha) {
-      this.pjson.scripts.test = `nyc ${this.ts ? '--extension .ts ' : ''}mocha --forbid-only "test/**/*.test.${this._ext}"`
+      this.pjson.scripts.test = `nyc ${this.ts ? '--extension .ts --require ts-node/register ' : ''}mocha --forbid-only "test/**/*.test.${this._ext}"`
```
Comment (amphro) [inline src/generators/app.ts:277]: I think this is typically in the mocha file, but shouldn't hurt.

**Review by amphro — APPROVED** [review]

**Pull request merged**